app/facades/menus/new_facade.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Menus
+  class NewFacade
+    def initialize(**params)
+      @menu = params[:menu]
+    end
+
+    def menu
+      @menu ||= Menu.new(meals: new_meals)
+    end
+
+    def menu_saved?
+      menu.persisted?
+    end
+
+    def food_items
+      @food_items ||= FoodItem.all
+    end
+
+    def weekend?
+      Date.today.saturday? || Date.today.sunday?
+    end
+
+    def menu_already_exist?
+      Menu.last&.date == Date.today
+    end
+
+    private
+
+    def new_meals
+      Array[new_meal]
```

Comment:
refactor `new_meals` and `new_meal` using memoization and methods without `new_` prefix.
Hint:
```ruby
def meal
  @meal ||= Meal.new
end
```